Added treatment protocol to schema - issue1428

ami-day opened the pull request:
### Release notes

Added a new treatment protocol.
 
### Reviews requested

- Need at least 2 Reviewers including from AIT/UCSC/Data Browser team to approve because this is a major update.
- Last Reviewer to review/approve, merge changes by 16/02/2022

### Background

At our recent wrangler catch-up with UCSC we talked about the necessity of a treatment_protocol entity to store information about how biomaterials have been treated with specific drugs / molecules, etc.
We currently do not have a location to store this information in our schema. It is blocking a dataset being wrangled, and we prioritised the metadata schema change to generate this new entity as a new tab in our wrangler spreadsheet. @hannes-ucsc , @theathorn , it would be useful to know how much work this would take for you to be able to do all of the steps necessary to represent it on the data portal (indexing, snapshotting, etc).

ami-day committed:
Added treatment_protocol.json

ami-day committed:
Added treatment protocol. #1428.

hannes-ucsc commented on json_schema/type/protocol/biomaterial_collection/treatment_protocol.json:
```diff
@@ -0,0 +1,85 @@
+            "description": "Method applied to cell culture to induce a specific treatment response.",
+            "type": "string",
+            "user_friendly": "Treatment method",
+            "comment": "Consider making this an ontology in the future.",
```
Why don't you do this now? Ontologizing properties in retroactively is a major effort, especially if the string values found in the wild aren't consistent with the ontology.

ESapenaVentura replied:
+1 to hannes, let's not leave to the future what we can do now just because we don't allocate capacity to research what terms could be used

ami-day replied:
Thanks for the feedback. I had left this comment after copying from an alternative json file, as I am actually not sure that we should require an ontology term for treatment method in particular.

An example could be: **T cell activation** (http://www.ebi.ac.uk/efo/EFO_0030037), which is a child of **assay** (http://purl.obolibrary.org/obo/OBI_0000070). I don't see many other treatment method examples however under assay or under the **experimental process** term (http://www.ebi.ac.uk/efo/EFO_0002694). What do you think? we could bring this up on Friday when we have a regular catch-up with the UCSC wranglers to make a decision.

ami-day replied:
I have added the new ontology.

ami-day committed:
Updated treatment_protocol.json

ami-day committed:
Added treatment method ontology

ami-day committed:
Updated versions.json

ami-day commented:
Hi @hannes-ucsc I have addressed the review comments and added an ontology to the treatment protocol schema, and made some other small updates to the pull request.

amnonkhen commented on json_schema/versions.json:
```diff
@@ -119,7 +119,8 @@
                     "enrichment_protocol": "3.1.0",
-                    "ipsc_induction_protocol": "3.2.0"
+                    "ipsc_induction_protocol": "3.2.0",
+                    "treatment_protocol": "0.0.0"
```
@ami-day Is `0.0.0` the correct value?

ami-day replied:
OH! no, it should be 0.0.1

hannes-ucsc replied:
Just noting that it wasn't actually changed to `0.0.1`

ami-day replied:
Sorry to get back so late on this, 0.0.0 is actually the correct value so I will leave it as it is

ami-day committed:
Corrected treatment protocol version

Corrected treatment protocol version

ami-day reviewed (commented):
Updated the treatment protocol schema version

ami-day commented:
Hi @hannes-ucsc @kbergin @aherbst-broad I have updated the ontology information and treatment protocol version. Is this pull request good to go? any other recommended changes? Thanks.

hannes-ucsc commented:
Some people don't get email notifications for GitHub activity. That's why the SOP includes weekly pings on Slack. The SOP also mandates that authors re-request reviews (here on GH, using the sidebar on the right) after they make changes, even from reviewers that have already approved.

ami-day committed:
Update json_schema/versions.json

Co-authored-by: ESapenaVentura <38617863+ESapenaVentura@users.noreply.github.com>

ami-day committed:
Update json_schema/type/protocol/biomaterial_collection/treatment_protocol.json

Co-authored-by: ESapenaVentura <38617863+ESapenaVentura@users.noreply.github.com>

ami-day commented on json_schema/type/protocol/biomaterial_collection/treatment_protocol.json:
```diff
@@ -0,0 +1,72 @@
+            },
+            "user_friendly": "Treatment reagents"
+        },
+        "target_pathway": {
+            "description": "Targeted pathway for specific treatment response.",
+            "type": "string",
+            "user_friendly": "Target pathway",
+            "example": "insulin signalling pathway; immune cell activation; chemokine signalling pathway."
```
Yes, I think this is a good idea!

ami-day replied:
Done.

ami-day committed:
Update json_schema/update_log.csv

Co-authored-by: ESapenaVentura <38617863+ESapenaVentura@users.noreply.github.com>

ami-day committed:
Update treatment_protocol.json

Added reference to module/ontology/target_pathway_ontology.json

ami-day committed:
Created target pathway ontology json

Created target pathway ontology file to be used with treatment method.

ami-day committed:
Updated update_log with new ontology modules

Updated update_log with new ontology modules

ami-day committed:
Updated versions.json with new  ontology modules

Updated versions.json with new  ontology modules

ESapenaVentura commented on json_schema/update_log.csv:
```diff
@@ -1 +1,4 @@
 Schema,Change type,Change message,Version,Date
+type/protocol/biomaterial_collection/treatment_protocol,major,"Added Treatment protocol entity. Fixes #1428",,
+module/ontology/treatment_method_ontology,major,"Added Treatment protocol method ontology. Fixes #1428",,
+module/ontology/target_pathway_ontology,major,"Added Treatment protocol target pathway ontology. Fixes #1428",,
```
Could you be a bit more specific with the change messages for each entity?

ami-day replied:
done

ESapenaVentura reviewed (changes requested):
Couple of minor comments/corrections, other than that LGTM!

Please remember that we need to add GO to HCAO and ensure that everything under `OBI:0000070` is in HCAO as well

ami-day committed:
Update json_schema/module/ontology/target_pathway_ontology.json

Co-authored-by: ESapenaVentura <38617863+ESapenaVentura@users.noreply.github.com>

ami-day committed:
Update json_schema/module/ontology/treatment_method_ontology.json

Co-authored-by: ESapenaVentura <38617863+ESapenaVentura@users.noreply.github.com>

ami-day committed:
Updated update_log

Updated update_log

ami-day commented:
Waiting to import the relevant ontology to HCAO before approving the pull request, will create a ticket in the ontology repo.

ami-day commented:
> Couple of minor comments/corrections, other than that LGTM!
> 
> Please remember that we need to add GO to HCAO and ensure that everything under `OBI:0000070` is in HCAO as well

Zoe is working on this!

ami-day commented:
Hi @hannes-ucsc, the version is actually correct (0.0.0) so I will leave it as it is, and have re-requested review. Thanks for your feedback!

hannes-ucsc reviewed (approved):
Updated list of reviewers to reflect the [recently updated list](https://github.com/HumanCellAtlas/dcp2/commit/2a3cfb1bb1efbe6b6c4ff73e8bcad100c91f928d) in the SOP.
